core/src/test/scala/unit/kafka/server/FetcherEventBusTest.scala

```diff
@@ -0,0 +1,182 @@
+package unit.kafka.server
+
+import kafka.server
+import kafka.server.{AddPartitions, DelayedFetcherEvent, FetcherEvent, FetcherEventBus, QueuedFetcherEvent, RemovePartitions, TruncateAndFetch}
+import kafka.utils.MockTime
+import org.apache.kafka.common.utils.Time
+import org.junit.Assert.{assertEquals, assertTrue, fail}
+import org.junit.Test
+
+import java.util.concurrent.{CountDownLatch, Executors}
+import scala.collection.Map
+import scala.collection.mutable.ArrayBuffer
+
+
+class FetcherEventBusTest {
+  @Test
+  def testGetWhileEmpty(): Unit = {
+    // test the getNextEvent method while the fetcherEventBus is empty
+    val fetcherEventBus = new FetcherEventBus(new MockTime())
+
+    val service = Executors.newSingleThreadExecutor()
+    @volatile var counter = 0
+    val runnableFinished = new CountDownLatch(1)
+    service.submit(new Runnable {
+      override def run(): Unit = {
+        // trying to call get will block indefinitely
+        fetcherEventBus.getNextEvent()
+        counter = 1
+        runnableFinished.countDown()
+      }
+    })
+
+    // sleep for 500ms, during which the runnable should still be blocked
+    Thread.sleep(500)
```

Comment:
Thanks for the advice. I've updated the PR and added a ConditionFactory to allow using a MockCondition in the test.
Despite that, I found the test may still take time due to the fact that kafka.utils.DelayedItem is using hard-coded Time.SYSTEM. Thus even if the new condition variable I added isn't really blocking in the test, the while loop will become a busy loop until the wall time says a delayed item has become current.

I'll just leave DelayedItem untouched now since changing it may trigger too many changes outside the scope of this PR.